x/staking/client/cli/query.go

```diff
@@ -197,22 +201,25 @@ $ gaiacli query staking delegations cosmos1gghjut3ccd8ay0zduzj64hwre2fxs9ld75ru9
 		RunE: func(cmd *cobra.Command, args []string) error {
 			cliCtx := context.NewCLIContext().WithCodec(cdc)
 
-			delegatorAddr, err := sdk.AccAddressFromBech32(args[0])
+			delAddr, err := sdk.AccAddressFromBech32(args[0])
 			if err != nil {
 				return err
 			}
 
-			resKVs, err := cliCtx.QuerySubspace(staking.GetDelegationsKey(delegatorAddr), storeName)
+			bz, err := cdc.MarshalJSON(staking.NewQueryDelegatorParams(delAddr))
```

Comment:
This could quite safely call `MustMarshalJSON` and avoid checking the error as the `NewQueryDelegatorParams()` call provides enough guarantees.

Reply:
Even though it's essentially "guaranteed" not to error, I'd rather avoid any panic-able situations in clients.

Reply:
As long as the type returned by `staking.NewQueryDelegatorParams()` is registered with the codec, this will never panic. To some extent, panic is desirable in this case because something absolutely wrong (i.e. something that goes by the definition of _unrecoverable error_) happened. I guess this is exactly the rationale behind all the `Must` prefixed function calls.

Reply:
Yeah in the state machine this is absolutely desirable!

> As long as the type ...

Reason enough to gracefully handle the error here :)

Reply:
I'm fine with this :+1: